### PR TITLE
Admin panel text. Initial creation

### PR DESCRIPTION
**Pull request by Tysyatsky**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

Task was about creating a text block for admin panel for further development.

## Summary of change

Added input fields and styles for them.

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Review by NadiaKishchuk — APPROVED** [review]
Good work

File: src/app/common/components/modals/Terms/DeleteTerm/DeleteTermModal.component.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            className="deleteModal"
+            closeIcon={<CancelBtn />}
+            open={deleteTerm.isOpen}
+            onCancel={() => setModal('deleteTerm')}
+            footer={[
+                <Button onClick={() => setModal('deleteTerm')}>Відміна</Button>,
+                <Button
+                    className="submit"
+                    onClick={() => {
+                        handleDelete(term?.id as number);
```
Comment (oshut) [inline src/app/common/components/modals/Terms/DeleteTerm/DeleteTermModal.component.tsx:31]: is casting really necessary here?

File: src/app/stores/term-store.ts [hunk 2]
```diff
@@ -39,27 +39,31 @@ export default class TermStore {
...
-                this.setItem(updatedTerm as Term);
-            });
+            if (id !== 0) {
+                await termsApi.update(id, term);
+                runInAction(() => {
+                    const updatedTerm = {
+                        ...this.TermMap.get(term.id),
+                        ...term,
+                    };
+                    this.setItem(updatedTerm as Term);
```
Comment (oshut) [inline src/app/stores/term-store.ts:51]: also casting here, do you really need it?

File: src/features/AdminPage/NewStreetcode/TextBlock/TextBlock.component.tsx [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import './TextBlock.styles.scss';
+
+import TextForm from './TextForm/TextForm.component';
+
+const TextBlock = () => (
+    <div className="text-block-container">
```
Comment (oshut) [inline src/features/AdminPage/NewStreetcode/TextBlock/TextBlock.component.tsx:6]: why some scss classes have '-' and other uses CamelCase?

File: src/features/AdminPage/NewStreetcode/TextBlock/TextForm/TextForm.component.tsx [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+    const [inputInfo, setInputInfo] = useState<Partial<InputInfoTextBlock>>();
+    const [showPreview, setShowPreview] = useState(false);
+    const { termsStore } = useMobx();
+    const { fetchTerms, getTermArray } = termsStore;
+    const [term, setTerm] = useState<Partial<Term>>();
+    const [selected, setSelected] = useState('');
+
+    useAsync(fetchTerms, []);
+
+    const maxTitleLenght = 50;
```
Comment (oshut) [inline src/features/AdminPage/NewStreetcode/TextBlock/TextForm/TextForm.component.tsx:31]: typo here
`length`

File: src/features/AdminPage/NewStreetcode/TextBlock/TextForm/TextForm.component.tsx [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+                    onChange={(e, editor) => {
+                        setInputInfo({ ...inputInfo, text: editor.getContent() });
+                    }}
+                    onSelectionChange={(e, editor) => {
+                        setSelected(editor.selection.getContent());
+                    }}
+                />
+                <Form.Item label="Оберіть пов'язаний термін">
+                    <Tooltip
+                        title={selected !== '' ? '' : 'Спочатку виділіть слово у тексті'}
```
Comment (oshut) [inline src/features/AdminPage/NewStreetcode/TextBlock/TextForm/TextForm.component.tsx:85]: I assume you can use double bang operator here : `!!`

File: src/features/AdminPage/NewStreetcode/TextBlock/TextForm/TextForm.component.tsx [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+                        selected !== '' && term !== undefined
+                            ? `${selected} з ${term?.title}` : 'Виділіть слово та термін!'
+                    }
+                    color="#8D1F16"
+                >
+                    <Button
+                        onClick={handleAddRelatedWord}
+                        disabled={selected === '' || term === undefined}
+                    >
+                        Пов&#39;язати
```
Comment (oshut) [inline src/features/AdminPage/NewStreetcode/TextBlock/TextForm/TextForm.component.tsx:112]: Is there any restrictions to use `'`?

File: src/features/AdminPage/NewStreetcode/TextBlock/TextForm/TextForm.component.tsx [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+            </Form.Item>
+            <Form.Item name="video" rules={[{ required: true, message: 'Будь ласка, введіть посилання!' }]}>
+                <div className="youtube-block">
+                    <h3>Відео</h3>
+                    <Input
+                        title="video"
+                        value={inputInfo?.link}
+                        className="smaller-input"
+                        placeholder="https://www.youtube.com"
+                        pattern="https?://www.youtube.com/watch.+"
```
Comment (oshut) [inline src/features/AdminPage/NewStreetcode/TextBlock/TextForm/TextForm.component.tsx:124]: What if pattern will be changes over a time? Consider to move it in some common place

File: src/features/AdminPage/NewStreetcode/TextBlock/TextForm/TextForm.component.tsx [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+                    </Button>
+                </Tooltip>
+            </Form.Item>
+            <Form.Item name="video" rules={[{ required: true, message: 'Будь ласка, введіть посилання!' }]}>
+                <div className="youtube-block">
+                    <h3>Відео</h3>
+                    <Input
+                        title="video"
+                        value={inputInfo?.link}
+                        className="smaller-input"
```
Comment (oshut) [inline src/features/AdminPage/NewStreetcode/TextBlock/TextForm/TextForm.component.tsx:122]: same thing with classes

File: src/features/AdminPage/NewStreetcode/TextBlock/TextForm/TextForm.component.tsx [hunk 9]
```diff
@@ -0,0 +1,167 @@
...
+                        name="link"
+                        required
+                        onChange={handleLinkChange}
+                    />
+                    <Tooltip
+                        title={
+                            inputInfo?.link?.includes('watch')
+                                ? '' : 'Вкажіть посилання на youtube.com/watch!'
+                        }
+                        color="#8D1F16"
```
Comment (oshut) [inline src/features/AdminPage/NewStreetcode/TextBlock/TextForm/TextForm.component.tsx:134]: you are using same color all over code, move it to some variable

File: src/features/AdminPage/NewStreetcode/TextBlock/TextForm/TextForm.component.tsx [hunk 10]
```diff
@@ -0,0 +1,167 @@
...
+                    </Tooltip>
+                    {
+                        inputInfo?.link?.includes('watch') && showPreview ? (
+                            <div>
+                                <h4>Попередній перегляд</h4>
+                                <iframe
+                                    title="video-preview"
+                                    src={
+                                        inputInfo?.link?.includes('/watch?v=')
+                                            ? inputInfo?.link?.replace('/watch?v=', '/embed/')
```
Comment (oshut) [inline src/features/AdminPage/NewStreetcode/TextBlock/TextForm/TextForm.component.tsx:151]: wouldn't it be better to have some private class encapsulating all logic?

File: src/features/AdminPage/TermDictionary/TermDictionary.component.tsx [hunk 11]
```diff
@@ -0,0 +1,138 @@
...
+    };
+
+    useEffect(() => {
+        fetchTerms();
+        setTableState();
+    }, []);
+
+    const handleAdd = () => {
+        const newTerm : Term = {
+            id: 0,
```
Comment (oshut) [inline src/features/AdminPage/TermDictionary/TermDictionary.component.tsx:39]: isn't Id is autoincremented?

**Pull request merged**